core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -571,22 +598,67 @@ public CacheStatusResponse doJoin(Address joiner, CacheJoinInfo joinInfo) throws
+                  return new CacheStatusResponse(null, currentTopology, stableTopology, availabilityMode);
+               }
+            } else {
+               if (isFirstMember) {
+                  // This node was the first to join. We need to install the initial CH
+                  CacheTopology initialTopology = createInitialCacheTopology();
+
+                  // Change our status
+                  status = ComponentStatus.RUNNING;
 
-            // Don't need to broadcast the initial CH update, just return the cache topology to the joiner
-            // But we do need to broadcast the initial topology as the stable topology
-            clusterTopologyManager.broadcastStableTopologyUpdate(cacheName, initialTopology, isTotalOrder(), isDistributed());
+                  // Don't need to broadcast the initial CH update, just return the cache topology to the joiner
+                  // But we do need to broadcast the initial topology as the stable topology
+                  clusterTopologyManager.broadcastStableTopologyUpdate(cacheName, initialTopology, isTotalOrder(), isDistributed());
+               }
+            }
          }
-         topologyBeforeRebalance = getCurrentTopology();
 
-         availabilityStrategy.onJoin(this, joiner);
+         topologyBeforeRebalance = getCurrentTopology();
+         // Only trigger availability strategy if we have a topology installed
+         if (topologyBeforeRebalance != null)
+            availabilityStrategy.onJoin(this, joiner);
       }
 
       return new CacheStatusResponse(null, topologyBeforeRebalance, stableTopology, availabilityMode);
    }
 
+   protected CacheTopology restoreCacheTopology(ScopedPersistentState state, Optional<Integer> persistentStateHash) {
+      if (trace) log.tracef("Attempting to restore CH for cache %s", cacheName);
+
+      try {
+         ConsistentHash persistedCH = joinInfo.getConsistentHashFactory().fromPersistentState(state);
+
+         if (!getExpectedMembers().containsAll(persistedCH.getMembers())) {
+            throw new IllegalStateException();
+         }
+         if (getExpectedMembers().size() > persistedCH.getMembers().size()) {
+            List<Address> extraneousMembers = new ArrayList<>(getExpectedMembers());
+            extraneousMembers.removeAll(persistedCH.getMembers());
+            throw log.extraneousMembersJoinRestoredCache(extraneousMembers, cacheName);
+         }
+         CacheTopology initialTopology = new CacheTopology(0, 0, persistedCH, null, persistedCH.getMembers());
+         setCurrentTopology(initialTopology);
+         setStableTopology(initialTopology);
+         rebalancingEnabled = true;
+         availabilityMode = AvailabilityMode.AVAILABLE;
+         return initialTopology;
+      } catch (IllegalStateException e) {
+         if (trace) log.tracef("Could not restore CH for cache %s", cacheName);
```

Comment:
I don't think it's a warning: most of the time this message will be printed as part of the normal join sequence: once for every node join until the original topology/CH can be restored.